### PR TITLE
Support custom registry and timeout

### PR DESCRIPTION
**Pull request by xpure-sklatt**
This provides 2 new options:

- **--registry** to set a custom registry url (including https and credentials)
- **--check-version-timeout** to set custom timeout value when checking for new version

Makes it possible to use custom npm registries that require https and user/password authenfication as well as a custom timeout value for fetching the new version numbers.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "next-update",
   "description": "Tests if module's dependencies can be updated to the newer version without breaking the tests",
-  "version": "0.0.0",
+  "version": "3.5.3",
```
Comment (bahmutov) [inline package.json]: please remove this - this module is using `semantic-release` to set the next version automatically

**Comment by bahmutov** [comment]
Great feature @xpure-sklatt but can you just remove the version from the package.json and I will merge it

**Comment by xpure-sklatt** [comment]
Cool, thanks @bahmutov! I've just reset the version to 0.0.0

**Comment by bahmutov** [comment]
Great, merged it in, hope to publish soon @xpure-sklatt

**Comment by xpure-sklatt** [comment]
Awesome, I'm looking forward to use it

**Pull request merged**